runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/MuleSdkOperationDoesNotHaveForbiddenFunctionsInExpressions.java

```diff
@@ -0,0 +1,99 @@
+
+  private final ExpressionLanguage expressionLanguage;
+
+  private static final String SINCE_PARAMETER = "since";
+  private static final String FUNCTION_PARAMETER = "function";
+  private static final MuleVersion DEPRECATE_UP_TO = new MuleVersion("2.5.0");
+  private static final Set<String> DEPRECATION_DOMAINS = ImmutableSet.of("dw", "Mule");
+
+  public MuleSdkOperationDoesNotHaveForbiddenFunctionsInExpressions(ExpressionLanguage el) {
+    this.expressionLanguage = el;
+  }
+
+  @Override
+  public String getName() {
+    return "Mule SDK Operation doesn't have deprecated expressions";
+  }
+
+  @Override
+  public String getDescription() {
+    return "The Mule SDK operation does not have a function as deprecated prior to DW 2.5 within itself";
+  }
+
+  private boolean isDeprecableProduct(ScopePhaseValidationItem warning) {
+    String product = warning.getParams().get(FUNCTION_PARAMETER).split("\\::")[0];
+    return DEPRECATION_DOMAINS.contains(product);
+  }
+
+  private boolean isDeprecatedVersion(ScopePhaseValidationItem warning) {
+    MuleVersion since = new MuleVersion(warning.getParams().get(SINCE_PARAMETER));
+    return since.sameAs(DEPRECATE_UP_TO) || since.priorTo(DEPRECATE_UP_TO);
+  }
+
+  private boolean isInvalidExpression(String expression) {
+    if (!isExpression(expression)) {
+      return false;
+    }
+    String actualExpression = getUnfixedExpression(expression);
+    ScopePhaseValidationMessages result =
+        expressionLanguage.collectScopePhaseValidationMessages(actualExpression, "", TypeBindings.builder().build());
+
+    return result.getWarnings().stream().anyMatch(warning -> warning.getKind().equals(DEPRECATED) && isDeprecatedVersion(warning)
+        && isDeprecableProduct(warning));
+  }
+
+  /**
+   * @return All the (raw) expressions within this {@link ParameterizedModel}
+   */
+  private Stream<String> getAllExpressions(ParameterizedModel model, ComponentAst componentAst) {
+    return model.getParameterGroupModels().stream()
+        .flatMap(groupModel -> groupModel.getParameterModels().stream()
+            .filter(parameter -> !parameter.getExpressionSupport().equals(ExpressionSupport.NOT_SUPPORTED))
+            .map(parameter -> componentAst.getParameter(groupModel.getName(), parameter.getName()).getRawValue())
+            .filter(Objects::nonNull));
+  }
+
+  @Override
+  public Optional<ValidationResultItem> validate(ComponentAst component, ArtifactAst artifact) {
+    if (component.recursiveStream().anyMatch(componentAst -> componentAst.getModel(ParameterizedModel.class)
+        .map(model -> getAllExpressions(model, componentAst).anyMatch(this::isInvalidExpression)).orElse(false))) {
+      return of(create(component, this,
+                       "Using an invalid function within a Mule SDK operation. All functions deprecated up to DataWeave 2.5 cannot be used inside a Mule Operation"));
```

Comment:
add the location of the invalid call within the script, refer to `AbstractExpressionSyntacticallyValid `